flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/optimize/program/FlinkChangelogModeInferenceProgram.scala

```diff
@@ -762,6 +757,81 @@ class FlinkChangelogModeInferenceProgram extends FlinkOptimizeProgram[StreamOpti
         Some(sink.copy(sinkTrait, children.head).asInstanceOf[StreamPhysicalRel])
       }
     }
+
```

Comment:
Can you add document here to explain `inferSinkRequiredTraits` and `analyzeUpsertMaterializeStrategy`?
1. .... should ...
2. .... should ...
3. ...